### PR TITLE
[Feature] add datset 2.0 support

### PR DESCRIPTION
**Pull request by wcchang1115**
Signed-off-by: wcchang <wcchang@infuseai.io>

### CODE REVIEW

File: primehub/datasets.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+        self.get(dataset_id)
+
+    def help_description(self):
+        return "Manage datasets"
+
+
+def validate_creation(config):
+    spec = ValidationSpec("""
+    input DatasetV2CreateInput {
+      id: String!
```
Comment (qrtt1) [inline primehub/datasets.py:372]: The id should be unique and not contain segment chars `.` or `/`
Suggest to follow other id rules like k8s names
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

**Review by qrtt1 — CHANGES_REQUESTED** [review]
should validate `id`

File: primehub/datasets.py (hunk 1 above)
Comment (wcchang1115) [reply primehub/datasets.py:382]: ok, add a validator

File: primehub/datasets.py [hunk 2]
```diff
@@ -385,3 +396,15 @@ def validate_update(config):
     }
     """)
     spec.validate(config)
+
+
+def validate_id(id: str):
```
Comment (qrtt1) [inline primehub/datasets.py:401]: Convert this to a new type for ValidationSpec: `ResourceId` or another better name

Comment (qrtt1) [reply primehub/datasets.py:401]: it shouldn't duplicate, just call the function from validation module

**Review by qrtt1 — CHANGES_REQUESTED** [review]

File: primehub/datasets.py (hunk 2 above)
Comment (wcchang1115) [reply primehub/datasets.py:401]: ok, will integrate with `validate_name` fuction in the validation module

**Review by qrtt1 — APPROVED** [review]

**Pull request merged**